src/LEGO.AsyncAPI/Validation/Rules/AsyncApiOperationRules.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) The LEGO Group. All rights reserved.
+
+using System.Linq;
+
+namespace LEGO.AsyncAPI.Validation.Rules
+{
+    using System;
+    using LEGO.AsyncAPI.Models;
+    using LEGO.AsyncAPI.Validations;
+
+    [AsyncApiRule]
+    public static class AsyncApiOperationRules
+    {
+        public static ValidationRule<AsyncApiOperation> OperationRequiredFields =>
+            new ValidationRule<AsyncApiOperation>(
+                (context, operation) =>
+                {
+                    context.Enter("action");
+                    if (operation.Action is null || !Enum.IsDefined(typeof(AsyncApiAction), operation.Action))
+                    {
+                        context.CreateError(
+                            nameof(OperationRequiredFields),
+                            string.Format(Resource.Validation_FieldRequired, "action", "operation"));
+                    }
+
+                    context.Exit();
+
+                    context.Enter("channel");
+                    if (operation.Channel?.Reference is null)
```

Comment:
Check channel directly here. No need to check the reference of it. It will always be a reference object